UNL-CSE-AMBASSADORS/VITA
# Cache busting

Wired up the NPM production command to 'cache bust' our JS and CSS files. Meaning that it appends a unique hash (of the contents of the file) to the end of the file name, and then replaces every instance of the original file path with the cache-busted version.

This is obviously useful so that users of the site don't have to "hard reload" when using the site. The only thing that this doesn't do is remove the "old" transpiled files from the /dist directory, which really shouldn't be a big deal, the solution would be to create a new temp folder, but I didn't think it was worth it. Let me know if you disagree and I can add that in real quick.

**TESTING**
* Run `npm install` and `npm update`
* Run `npm run dev` (or the ./script/build.sh file) 
  * Ensure the website still works with no console errors / missing file errors / style errors
* Run `npm run production`, and you will notice that it replaces instances of our `/dist/**/*.js/css` files with `/dist/**/*-HASH.js/css`
  * Ensure the website still works with no console errors / missing file errors / style errors
  * Run `git reset HEAD --hard` to return to the state before the production build changes
